source/lac/petsc_solver.cc

```diff
@@ -599,14 +583,16 @@ namespace PETScWrappers
   void
   SolverLSQR::set_solver_type(KSP &ksp) const
   {
-    PetscErrorCode ierr = KSPSetType(ksp, KSPLSQR);
-    AssertThrow(ierr == 0, ExcPETScError(ierr));
+    AssertPETSc(KSPSetType(ksp, KSPLSQR));
 
     // in the deal.II solvers, we always
     // honor the initial guess in the
     // solution vector. do so here as well:
-    ierr = KSPSetInitialGuessNonzero(ksp, PETSC_TRUE);
-    AssertThrow(ierr == 0, ExcPETScError(ierr));
+    AssertPETSc(KSPSetInitialGuessNonzero(ksp, PETSC_TRUE));
+
+    // The KSPLSQR implementation keeps overwriting the user-defined
+    // convergence test.
+    ensure_convergence_test();
```

Comment:
:-) Is this a PETSc bug? (If so, maybe say so.)

Reply:
Yep, for me the related PETSc code is bugged. Instead, for the author of the offending PETSc lines, it is apparently a design decision. Derails of a welcoming community for open-source software ....

Reply:
added some more description